pkgs/by-name/sh/shadps4/package.nix

```
@@ -0,0 +1,79 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  xorg,
+  qt6,
+  vulkan-headers,
+  vulkan-loader,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "shadps4";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "shadps4-emu";
+    repo = "shadPS4";
+    rev = "refs/tags/v.${finalAttrs.version}";
+    hash = "sha256-HNO/L2cMZVHTHczjKgUFVDDdWBnj7SKctGi+P7jATpQ=";
+    fetchSubmodules = true;
+  };
+
+  patches = [ ./bloodborne.patch ]; # https://github.com/shadps4-emu/shadPS4/issues/758
+
+  buildInputs = [
+    xorg.libX11
+    xorg.libXext
+    qt6.qtbase
+    qt6.qtdeclarative
+    qt6.qtwayland
+    vulkan-headers
+    vulkan-loader
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    qt6.wrapQtAppsHook
+  ];
+
+  postPatch = ''
+    substituteInPlace src/common/path_util.cpp \
+      --replace-fail 'const auto user_dir = std::filesystem::current_path() / PORTABLE_DIR;' \
+      'const auto user_dir = std::filesystem::path(getenv("HOME")) / ".config" / "shadPS4";'
+  '';
+
+  configurePhase = ''
+    cmake -B build -DCMAKE_BUILD_TYPE=DEBUG -DENABLE_QT_GUI=ON
+  '';
+
+  buildPhase = ''
+    cmake --build build --config Release --parallel
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D -t $out/bin build/shadps4
+    install -Dm644 -t $out/share/icons/hicolor/512x512/apps $src/.github/shadps4.png
+    install -Dm644 -t $out/share/applications $src/.github/shadps4.desktop
+
+    runHook postInstall
+  '';
+
+  fixupPhase = ''
+    patchelf --add-rpath ${lib.makeLibraryPath [ vulkan-loader ]} \
+      $out/bin/shadps4
+  '';
+
+  meta = {
+    description = "Early in development PS4 emulator";
+    homepage = "https://github.com/shadps4-emu/shadPS4";
+    changelog = "https://github.com/shadps4-emu/shadPS4/releases/tag/v.${finalAttrs.version}";
+    license = lib.licenses.gpl2Only;
+    maintainers = with lib.maintainers; [ ryand56 ];
+    mainProgram = "shadps4";
+    platforms = lib.platforms.linux;
```

On aarch64-linux:
```
In file included from /build/source/src/core/libraries/libc/printf.h:65,
                 from /build/source/src/core/libraries/libc/libc_stdio.h:7,
                 from /build/source/src/core/libraries/libc/libc_stdio.cpp:7:
/build/source/src/core/libraries/libc/va_ctx.h:4:10: fatal error: xmmintrin.h: No such file or directory
    4 | #include <xmmintrin.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
```

So at minimum:

```suggestion
    platforms = lib.intersectLists lib.platforms.linux lib.platforms.x86;
```

Though prolly `lib.platforms.x86_64`, I don't think it'll work on 32-bit x86.